plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,614 @@
+    params = {}
+    params['Bucket'] = bucket_name
+
+    object_legal_lock_configuration_info = {}
+
+    try:
+        object_legal_lock_configuration_info = connection.get_object_lock_configuration(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        object_legal_lock_configuration_info['msg'] = 'Object lock configuration not found, are you sure the bucket/object configuration exist?'
+
+    if len(object_legal_lock_configuration_info) != 0 and 'msg' not in object_legal_lock_configuration_info.keys():
+        # Remove ResponseMetadata from object_legal_lock_configuration_info, convert to snake_case
+        del(object_legal_lock_configuration_info['ResponseMetadata'])
+        object_legal_lock_configuration_info = camel_dict_to_snake_dict(object_legal_lock_configuration_info)
+
+    return object_legal_lock_configuration_info
+
+
+def describe_s3_object_retention(connection, module, bucket_name, object_name):
+    params = {}
+    params['Bucket'] = bucket_name
+    params['Key'] = object_name
+
+    object_retention_info = {}
+
+    try:
+        object_retention_info = connection.get_object_retention(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        object_retention_info['msg'] = 'Object retention info not found, are you sure the bucket/object configuration exist?'
+
+    if len(object_retention_info) != 0 and 'msg' not in object_retention_info.keys():
+        # Remove ResponseMetadata from object_retention_info, convert to snake_case
+        del(object_retention_info['ResponseMetadata'])
+        object_retention_info = camel_dict_to_snake_dict(object_retention_info)
+
+    return object_retention_info
+
+
+def describe_s3_object_tagging(connection, module, bucket_name, object_name):
+    params = {}
+    params['Bucket'] = bucket_name
+    params['Key'] = object_name
+
+    object_tagging_info = {}
+
+    try:
+        object_tagging_info = connection.get_object_tagging(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        object_tagging_info['msg'] = 'Object tags not found, are you sure the bucket/object tags exist?'
+
+    if len(object_tagging_info) != 0 and 'msg' not in object_tagging_info.keys():
+        # Remove ResponseMetadata from object_tagging_info, convert to snake_case
+        del(object_tagging_info['ResponseMetadata'])
+        object_tagging_info = boto3_tag_list_to_ansible_dict(object_tagging_info['TagSet'])
+
+    return object_tagging_info
+
+
+def get_object_details(connection, module, bucket_name, object_name, requested_facts):
+
+    all_facts = {}
+
+    # Remove non-requested facts
+    requested_facts = {fact: value for fact, value in requested_facts.items() if value is True}
+
+    all_facts['object_metadata'] = get_object(connection, module, bucket_name, object_name)['object_metadata']
```

Comment:
I won't probably call it `object_metadata`, it is probably misleading, but I do not have a strong opinion about it.

Reply:
that makes sense, not entirely sure about what could be a good name for the output param here though, considering the data it returns is from https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.get_object